fv3core/stencils/updatedzd.py

```diff
@@ -151,154 +256,144 @@ def out(zs: sd, zh: sd, ws: sd, dt: float):
             zh[0, 0, 0] = zh if zh > other else other
 
 
-def compute(ndif, damp_vtd, dp0, zs, zh, crx, cry, xfx, yfx, wsd, dt):
+def compute(
+    ndif: FloatField,
+    damp_vtd: FloatField,
+    dp0: FloatField,
+    zs: FloatField,
+    zh: FloatField,
+    crx: FloatField,
+    cry: FloatField,
+    xfx: FloatField,
+    yfx: FloatField,
+    wsd: FloatField,
+    dt: float,
+):
     grid = spec.grid
-
-    ndif[-1] = ndif[-2]
-    damp_vtd[-1] = damp_vtd[-2]
+    halo = grid.halo
 
     crx_adv = utils.make_storage_from_shape(
-        crx.shape, grid.compute_origin(add=(0, -grid.halo, 0))
+        crx.shape, grid.compute_origin(add=(0, -halo, 0))
     )
     cry_adv = utils.make_storage_from_shape(
-        cry.shape, grid.compute_origin(add=(-grid.halo, 0, 0))
+        cry.shape, grid.compute_origin(add=(-halo, 0, 0))
     )
     xfx_adv = utils.make_storage_from_shape(
-        xfx.shape, grid.compute_origin(add=(0, -grid.halo, 0))
+        xfx.shape, grid.compute_origin(add=(0, -halo, 0))
     )
     yfx_adv = utils.make_storage_from_shape(
-        yfx.shape, grid.compute_origin(add=(-grid.halo, 0, 0))
+        yfx.shape, grid.compute_origin(add=(-halo, 0, 0))
     )
     ra_x = utils.make_storage_from_shape(
-        crx.shape, grid.compute_origin(add=(0, -grid.halo, 0))
+        crx.shape, grid.compute_origin(add=(0, -halo, 0))
     )
     ra_y = utils.make_storage_from_shape(
-        cry.shape, grid.compute_origin(add=(-grid.halo, 0, 0))
+        cry.shape, grid.compute_origin(add=(-halo, 0, 0))
     )
 
-    gam = utils.make_storage_from_shape(zs.shape, grid.full_origin())
-    edge_profile(
+    edge_profile_stencil(
         crx,
         xfx,
         crx_adv,
         xfx_adv,
-        dp0,
-        gam,
-        origin=(grid.is_, grid.jsd, 0),
-        domain=(grid.nic + 1, grid.njd, grid.npz + 1),
-    )
-    # edge_python(crx, xfx, crx_adv, xfx_adv, dp0, gam, slice(grid.is_, grid.ie
-    # + 2), slice(grid.jsd, grid.jed+1),  qe1_2, gam_2)
-
-    gam = utils.make_storage_from_shape(zs.shape, grid.full_origin())
-    edge_profile(
         cry,
         yfx,
         cry_adv,
         yfx_adv,
         dp0,
-        gam,
-        origin=(grid.isd, grid.js, 0),
-        domain=(grid.nid, grid.njc + 1, grid.npz + 1),
+        origin=grid.compute_origin(add=(-halo, -halo, 0)),
+        domain=grid.domain_shape_compute(add=(2 * halo, 2 * halo, 1)),
     )
-
     ra_stencil_update(
         grid.area,
         xfx_adv,
-        ra_x,
         yfx_adv,
+        ra_x,
         ra_y,
-        origin=grid.compute_origin(add=(-grid.halo, -grid.halo, 0)),
-        domain=(grid.nid, grid.njd, grid.npz + 1),
+        origin=grid.compute_origin(add=(-halo, -halo, 0)),
+        domain=grid.domain_shape_compute(add=(2 * halo, 2 * halo, 1)),
```

Comment:
this one probably should be grid.full_origin() and grid.domain_shape_full() though, i think, because they come up quite often. if you disagree I am open to discuss